secboot/secboot_tpm.go

```diff
@@ -338,10 +346,16 @@ func SealKey(key EncryptionKey, params *SealKeyParams) error {
 
 	// Seal key to the TPM
 	creationParams := sb.KeyCreationParams{
-		PCRProfile: pcrProfile,
-		PINHandle:  pinHandle,
+		PCRProfile:             pcrProfile,
+		PCRPolicyCounterHandle: policyCounterHandle,
+	}
+
+	authKey, err := sbSealKeyToTPM(tpm, key[:], params.KeyFile, &creationParams)
+	if err := osutil.AtomicWriteFile(params.TPMPolicyAuthKeyFile, authKey, 0600, 0); err != nil {
+		return fmt.Errorf("cannot write the policy auth key file: %v", err)
 	}
-	return sbSealKeyToTPM(tpm, key[:], params.KeyFile, params.TPMPolicyUpdateDataFile, &creationParams)
+
+	return err
```

Comment:
(nitpick) We handled the error conditions before AFAICT so we can return "nil" here I think.

Reply:
mmh, I get a test error if I change this, digging

Reply:
actually the code was strange, there was an actual bug